clang/include/clang/3C/ConstraintsGraph.h

```diff
@@ -163,7 +167,10 @@ class DataGraph
     else
       Edges = N->getPredecessors();
```

Comment:
I am wondering if there's a glitch for when you get edges via predecessors vs. via successors. When you run with your debugging statements below, do you see the skipping of soft edges happening as you'd expect?